scipy/linalg/decomp_svd.py

```diff
@@ -264,35 +264,101 @@ def diagsvd(s, M, N):
 
 # Orthonormal decomposition
 
-def orth(A):
+def orth(A, rcond=None):
     """
     Construct an orthonormal basis for the range of A using SVD
 
     Parameters
     ----------
     A : (M, N) array_like
         Input array
+    rcond : float, optional
+        Relative condition number. Singular values ``s`` smaller than
+        ``rcond * max(s)`` are considered zero.
+        Default: floating point eps * max(M,N).
 
     Returns
     -------
     Q : (M, K) ndarray
         Orthonormal basis for the range of A.
-        K = effective rank of A, as determined by automatic cutoff
+        K = effective rank of A, as determined by rcond
 
     See also
     --------
     svd : Singular value decomposition of a matrix
+    null_space : Matrix null space
 
     """
     u, s, vh = svd(A, full_matrices=False)
-    M, N = A.shape
-    eps = numpy.finfo(float).eps
-    tol = max(M, N) * numpy.amax(s) * eps
+    M, N = u.shape[0], vh.shape[1]
+    if rcond is None:
+        rcond = numpy.finfo(float).eps * max(M, N)
```

Comment:
If our `svd` functions accept (and return) both `float32` and `float64` types, wouldn't these ideally be `finfo(s.dtype)` or so? I guess we can't change it for backward compatibility reasons even if so.

Reply:
Yes, I think you're right it should be s.dtype here.

I would maybe argue that it is a bug that it's not, so we can skip the backward compatibility issue. Namely, already the following fails,
```
>>> a = np.ones((5, 5), dtype=np.float32)
>>> scipy.linalg.orth(a).shape
(5, 2)
```
Certainly the range should be 1-dimensional...